### DISCUSSION TITLE
Does it make sense to provide the pageContext via a React Context from a wrapPageElement?

### DISCUSSION THREAD

**Discussion by openscript** [gatsbyjs/gatsby#33569] [Help] [answered]
Dear Gatsby community

Sometimes I need Gatsbys `pageContext` in a deeply nested component inside the page. Does something speak against providing the `pageContext` via a React Context from a wrapPageElement?

Hopefully it's okay to post this question here. I'm looking forward to hearing from you.

Robin

**Comment by herecydev** [comment, marked as answer]
I do exactly the same thing, it seems like a great use of React context.
